tools/search_hadith.py

```diff
@@ -13,14 +13,14 @@ def get_tool_description(self):
         return {
             "type": "function",
             "function": {
-                "name": TOOL_NAME,
-                "description": "Search the Hadith for relevant narrations. Returns a list of hadith. Multiple hadith may be relevant.",
+                "name": "search_hadith",
+                "description": "Search for relevant Hadith narrations based on a specific topic.",
```

Comment:
Why the change?

Reply:
gpt-4o was deciding to use the tools but wasn't outputting the required arguments (the `query` parameter in this case). This was causing the CI check to fail. After modifying the tool description, the model started providing the arguments correctly.